janus_core/cli/phonons.py

```diff
@@ -0,0 +1,295 @@
+    ] = 0.0,
+    temp_end: Annotated[
+        float,
+        Option(help="End temperature for thermal properties calculations, in K."),
+    ] = 1000.0,
+    temp_step: Annotated[
+        float,
+        Option(help="Temperature step for thermal properties calculations, in K."),
+    ] = 50,
+    band: Annotated[
+        bool,
+        Option(help="Whether to compute band structure."),
+    ] = False,
+    hdf5: Annotated[
+        bool,
+        Option(help="Whether to save force constants in hdf5."),
+    ] = True,
+    plot_to_file: Annotated[
+        bool,
+        Option(help="Whether to plot bandstructure and/pr dos/pdos when calculated."),
+    ] = False,
+    dos: Annotated[
+        bool,
+        Option(help="Whether to calculate the DOS."),
+    ] = False,
+    pdos: Annotated[
+        bool,
+        Option(
+            help="Whether to calculate the PDOS.",
+        ),
+    ] = False,
+    minimize: Annotated[
+        bool,
+        Option(
+            help="Whether to minimize structure before calculations.",
+        ),
+    ] = False,
+    symmetrize: Annotated[
+        bool, Option(help="Whether to symmetrize force constants.")
+    ] = False,
+    fmax: Annotated[
+        float, Option(help="Maximum force for optimization convergence.")
+    ] = 0.1,
+    minimize_kwargs: MinimizeKwargs = None,
+    arch: Architecture = "mace_mp",
+    device: Device = "cpu",
+    read_kwargs: ReadKwargs = None,
+    calc_kwargs: CalcKwargs = None,
+    file_prefix: Annotated[
+        Optional[Path],
+        Option(
+            help=(
+                """
+                Prefix for output filenames. Default is inferred from structure name,
+                or chemical formula.
+                """
+            ),
+        ),
+    ] = None,
+    log: LogPath = "phonons.log",
+    summary: Summary = "phonons_summary.yml",
+):
+    """
+    Perform phonon calculations and write out results.
+
+    Parameters
+    ----------
+    ctx : Context
+        Typer (Click) Context. Automatically set.
+    struct : Path
+        Path of structure to simulate.
+    struct_name : Optional[PathLike]
+        Name of structure to simulate. Default is inferred from filepath or chemical
+        formula.
+    supercell : str
+        Supercell lattice vectors. Must be passed in the form '1x2x3'. Default is
+        2x2x2.
+    displacement : float
+        Displacement for force constants calculation, in A. Default is 0.01.
+    thermal : bool
+        Whether to calculate thermal properties. Default is False.
+    temp_start : float
+        Start temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 0.0.
+    temp_end : float
+        End temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 1000.0.
+    temp_step : float
+        Temperature step for CV calculations, in K. Unused if `thermal` is False.
```

Comment:
Do these need to match the definitions above?

Reply:
what do you mean?

Reply:
Above it's swapped from "CV calculations" to "thermal properties calculations"

Reply:
fair point I will harmonise...